core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1973,6 +1981,7 @@ public static ExpressionAnalysis analyzeExpression(
         analyzer.analyze(expression, scope);
 
         updateAnalysis(analysis, analyzer, session, accessControl);
+        analysis.addFieldLineage(expression, analyzer.getSourceFields());
```

Comment:
There are other method which should be updated, see `analyzeWindow` or `analyzeExpressions` where this is not added.